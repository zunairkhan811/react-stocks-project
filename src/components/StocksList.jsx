import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import fetchStocksData from '../redux/stocks/fetchApi';
import { useState } from 'react';

const StocksList = () => {
  const [searchText, setSearchText] = useState('');
  const dispatch = useDispatch();
  const { data, message, isLoading, isError } = useSelector((store) => store.stocksList);
  console.log(data);
  useEffect(() => {
    if (data.length === 0) {
      dispatch(fetchStocksData());
    }
  }, [dispatch,data]);

  if(isLoading){
    return(
      <div className='list-heading-container'>
        <h1>{message}</h1>
      </div>
    )
  }
  if(isError){
    return(
      <div className='list-heading-container'>
        <h1>{message}</h1>
      </div>
    )
  }

  return (
    <>
      <div className='list-main-container'>
        <div className='list-heading-container'>
          <h1>NASDAQ Stocks</h1>
          <input type="text" placeholder='Search with symbol...' onChange=
          {(e)=>setSearchText(e.target.value)} />
        </div>
        <div className='list-container'>
          {data.filter((item)=>{
              if(searchText === ''){
                return item;
              }
              if(item.symbol.toLowerCase().includes(searchText.toLowerCase())){
                return item;
              }
            })
              .map((element)=>(
              
                <div key={element.symbol} className='list-items'>
                  <h1>Symbol - {element.symbol}</h1>
                  <h2>Company Name - {element.companyName}</h2>
                </div>

              ))
          }
        </div>
        

      </div>
      
    </>
    
  );
};

export default StocksList;
