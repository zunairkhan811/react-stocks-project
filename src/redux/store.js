import { configureStore } from '@reduxjs/toolkit';
import rootReducer from './rootreducer';

const store = configureStore({
  reducer: rootReducer,
});

export default store;
