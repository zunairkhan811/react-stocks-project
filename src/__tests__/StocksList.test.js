import renderer from "react-test-renderer";
import StocksList from "../components/StocksList";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import store from "../redux/store";
import '@testing-library/jest-dom';

it('StockList renders correctly',()=>{
    const tree = renderer.create(
        
        <BrowserRouter>
        <Provider store={store}>
        <StocksList />
        </Provider>
            
        </BrowserRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
})