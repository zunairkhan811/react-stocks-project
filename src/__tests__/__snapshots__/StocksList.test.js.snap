// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockList renders correctly 1`] = `
<div
  className="list-main-container"
>
  <div
    className="list-heading-container"
  >
    <h1>
      NASDAQ Stocks
    </h1>
    <input
      onChange={[Function]}
      placeholder="Search with symbol..."
      type="text"
    />
  </div>
  <div
    className="list-container"
  />
</div>
`;
